### PR TITLE
[emr_add_steps_operator] Allow job_flow_id & job_flow_name (Priority is in the order of id, name.)

### PR DESCRIPTION
**Pull request by seunggabi**
related #31314

---
**^ Add meaningful description above**
- I need to job_flow_id & job_flow_name (for choosing in runtime).
- So, I suggest this code. Thx for review.

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/providers/amazon/aws/operators/emr.py [hunk 1]
```diff
@@ -87,9 +87,14 @@ def __init__(
         execution_role_arn: str | None = None,
         **kwargs,
     ):
-        if not exactly_one(job_flow_id is None, job_flow_name is None):
-            raise AirflowException("Exactly one of job_flow_id or job_flow_name must be specified.")
+        if (job_flow_id is None) and (job_flow_name is None):
+            raise AirflowException(
+                "Exactly job_flow_id or job_flow_name must be existed. "
```
Comment (pierrejeambrun) [inline airflow/providers/amazon/aws/operators/emr.py]:
```suggestion
                 "At least one of job_flow_id or job_flow_name must be specified. "
```

If both are specified maybe we should log something to warn that something might not work as expected. (id takes precedence)

File: tests/providers/amazon/aws/operators/test_emr_add_steps.py [hunk 2]
```diff
@@ -83,12 +83,14 @@ def test_init(self):
     @pytest.mark.parametrize(
         "job_flow_id, job_flow_name",
         [
-            pytest.param("j-8989898989", "test_cluster", id="both-specified"),
             pytest.param(None, None, id="both-none"),
         ],
     )
     def test_validate_mutually_exclusive_args(self, job_flow_id, job_flow_name):
-        error_message = r"Exactly one of job_flow_id or job_flow_name must be specified\."
+        error_message = (
```
Comment (pierrejeambrun) [inline tests/providers/amazon/aws/operators/test_emr_add_steps.py]: Name of the test is not relevant anymore, we need to adapt.

**Review by pierrejeambrun — COMMENTED** [review]
I'm not a big fan of this change, I prefer an explicit error rather than silently giving priority to one of the two parameters. This could cause hard to debug situations.

Curious to see what others think.

**Comment by vincbeck** [comment]
I agree with @pierrejeambrun. With this change, when both parameters are provided, `job_flow_name` will be ignored. It can be really confusing for the user to pass some parameter which at the end, is not used. I am not strongly opinionated on this one though, so if you feel there is a real need there, go for it. But if so, we should definitely update the doc (`docs/apache-airflow-providers-amazon/operators/emr/emr.rst`) to explicitly explain this behavior.

File: tests/providers/amazon/aws/operators/test_emr_add_steps.py (hunk 2 above)
Comment (seunggabi) [reply tests/providers/amazon/aws/operators/test_emr_add_steps.py]: thx. I changed this.

Comment (pierrejeambrun) [reply tests/providers/amazon/aws/operators/test_emr_add_steps.py]: You can also remove the parametrise if there is only one set of parameters to test.

**Pull request closed without merging**